tests/h/services/user_signup_test.py

```diff
@@ -165,75 +166,34 @@ def test_signup_raises_conflict_error_when_account_already_exists(
-    @pytest.fixture
-    def mailer(self):
-        return mock.Mock(spec_set=["send"])
+    @pytest.fixture(autouse=True)
+    def tasks_mailer(self, patch):
+        return patch("h.services.user_signup.tasks_mailer")
 
-    @pytest.fixture
-    def signup_email(self):
-        signup_email = mock.Mock(spec_set=[])
-        signup_email.return_value = (
-            ["test@example.com"],
-            "My subject",
-            "Text",
-            "<p>HTML</p>",
-        )
-        return signup_email
+    @pytest.fixture(autouse=True)
+    def signup(self, patch):
+        return patch("h.services.user_signup.signup")
 
 
 @pytest.mark.usefixtures("user_password_service")
 class TestUserSignupServiceFactory:
-    def test_returns_user_signup_service(self, pyramid_request):
-        svc = user_signup_service_factory(None, pyramid_request)
-
-        assert isinstance(svc, UserSignupService)
-
-    def test_provides_request_default_authority_as_default_authority(
-        self, pyramid_request
-    ):
-        svc = user_signup_service_factory(None, pyramid_request)
-
-        assert svc.default_authority == pyramid_request.default_authority
+    def test_it(self, UserSignupService, pyramid_request, user_password_service):
+        svc = user_signup_service_factory(sentinel.context, pyramid_request)
 
-    def test_provides_request_db_as_session(self, pyramid_request):
-        svc = user_signup_service_factory(None, pyramid_request)
-
-        assert svc.session == pyramid_request.db
-
-    def test_wraps_email_module_as_signup_email(self, patch, pyramid_request):
-        signup_email = patch("h.emails.signup.generate")
-        svc = user_signup_service_factory(None, pyramid_request)
-
-        svc.signup_email(user_id=123, email="foo@bar.com", activation_code="abc456")
-
-        signup_email.assert_called_once_with(
-            pyramid_request, user_id=123, email="foo@bar.com", activation_code="abc456"
+        UserSignupService.assert_called_once_with(
+            request=pyramid_request,
+            default_authority=pyramid_request.default_authority,
+            password_service=user_password_service,
         )
+        assert svc == UserSignupService.return_value
 
-    def test_provides_user_password_service(self, pyramid_request):
-        svc = user_signup_service_factory(None, pyramid_request)
-        password_svc = pyramid_request.find_service(name="user_password")
-
-        assert svc.password_service == password_svc
-
-
-@pytest.fixture
-def user_password_service(pyramid_config):
-    service = mock.Mock(spec_set=UserPasswordService())
-
-    def password_setter(user, password):  # pylint:disable=unused-argument
-        user.password = "fakehash"
-
-    service.update_password.side_effect = password_setter
-
-    pyramid_config.register_service(service, name="user_password")
-    return service
+    @pytest.fixture
+    def UserSignupService(self, patch):
+        return patch("h.services.user_signup.UserSignupService")
```

Comment:
This diff is a giant mess, but the original test for the factory was in the style where we had a different test for each argument we passed to the service. 

Allow me a quite whine. This is too much and I'd say is actively worse than a single test for a few reasons:

 * General bloat - the tests are out of proportion to the code tested
 * With a single test you have to get the call right, with many you can forget a bit
 * It's hard to look at the tests and see what they are testing

Other oddities here include mocking things we shouldn't, having active mocks which try to "do" things rather than relying on `return_value` and not mocking the actual service.

If we want to assert behavior about the init of the service I think that should go in the service tests?